iceoryx_posh/source/popo/waitset/wait_set.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+#include "iceoryx_posh/internal/popo/waitset/wait_set.hpp"
+
+namespace iox
+{
+namespace popo
+{
+WaitSet::WaitSet(cxx::not_null<ConditionVariableData* const> condVarDataPtr) noexcept
+    : m_conditionVariableDataPtr(condVarDataPtr)
+    , m_conditionVariableWaiter(m_conditionVariableDataPtr)
+    , m_guardCondition(m_conditionVariableDataPtr)
+{
+    m_conditionVector.push_back(&m_guardCondition);
+}
+
+bool WaitSet::attachCondition(Condition& condition) noexcept
+{
+    if (!condition.isConditionVariableAttached())
+    {
+        if (condition.attachConditionVariable(m_conditionVariableDataPtr))
+        {
+            return m_conditionVector.push_back(&condition);
+        }
+    }
+
+    return false;
+}
+
+bool WaitSet::detachCondition(const Condition& condition) noexcept
+{
+    for (auto currentCondition : m_conditionVector)
```

Comment:
We are working on copies instead of references here. Please write `auto & currentCondition : m_conditionVector`

Reply:
Done